pint/compat.py

```diff
@@ -303,7 +303,10 @@ def eq(lhs: Any, rhs: Any, check_all: bool) -> Union[bool, Iterable[bool]]:
     -------
     bool or array_like of bool
     """
-    out = lhs == rhs
+    try:
+        out = lhs == rhs
+    except ValueError:
+        return False
```

Comment:
What is the reason for this change? Two things being incomparable is not necessarily the same as not being equal. This is a behavioural change that I don't see how it is related to the bug to be fixed (except maybe to simplify the test code?). This may potentially silently cause wrong behaviour. That said, the contract for `eq` isn't really clear and I don't know how/where it is used. It is well possible that the behaviour before the change was wrong.

Reply:
my mistake, this was to get something working in pint-pandas. reverted this